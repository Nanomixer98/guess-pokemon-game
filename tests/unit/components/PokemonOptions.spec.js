import { shallowMount } from '@vue/test-utils';
import PokemonOptions from '@/components/PokemonOptions';
import { pokemons } from '../mocks/pokemons.mock';

describe('PokemonOptions', () => {

    let wrapper;

    beforeEach(() => {
        wrapper = shallowMount(PokemonOptions, {
            props: {
                pokemons
            }
        })
    })

    test('debe de hacer match con el snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })

    test('debe de mostrar las 4 opciones correctamente', () => {
        const liTags = wrapper.findAll('li');
        expect(liTags.length).toBe(4);

        expect(liTags[0].text()).toBe('bulbasaur');
        expect(liTags[1].text()).toBe('ivysaur');
        expect(liTags[2].text()).toBe('mew');
        expect(liTags[3].text()).toBe('charmander');
    })

    test('debe de emitir "selection" con sus respectivos parametros al hacer click', () => {
        const [li1, li2, li3, li4] = wrapper.findAll('li');

        li1.trigger('click')
        li2.trigger('click')
        li3.trigger('click')
        li4.trigger('click')
        console.log(wrapper.emitted('selectedPokemon'));
        expect(wrapper.emitted('selectedPokemon').length).toBe(4)
        expect(wrapper.emitted('selectedPokemon')[0]).toEqual([4])
        expect(wrapper.emitted('selectedPokemon')[1]).toEqual([5])
        expect(wrapper.emitted('selectedPokemon')[2]).toEqual([7])
        expect(wrapper.emitted('selectedPokemon')[3]).toEqual([12])
    })

})