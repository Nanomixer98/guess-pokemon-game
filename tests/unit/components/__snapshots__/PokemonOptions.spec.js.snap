// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOptions debe de hacer match con el snapshot 1`] = `
<div class="options-container">
  <ul>
    <li>bulbasaur</li>
    <li>ivysaur</li>
    <li>mew</li>
    <li>charmander</li>
  </ul>
</div>
`;
